### PR TITLE
General cleanup

### LINKED ISSUE #247: Consolidate client.on('ready') functions
There are two functions that handle the same `client.on('ready')` event - one in `app.js`, and one in `ready.js`. These should be consolidated into `ready.js`.

### PR DESCRIPTION
**Pull request by debashisbiswas**
Here's a bunch of small changes around the codebase that we've been talking about updating for a while. Well, there's a lot, so it's not very small anymore, but...

### Significant changes
- delete raid command and the corresponding test

### Bug fixes
- fix array out of bounds error in `article.js`
- handle case in `cotw.js` for when the data is undefined
- fix issue in which roles were not being assigned in `cotwActions.js`
- fix bug in `cotwActions.js` that caused words connected to newlines to be identified as part of a poll or challenge link
- updated control flow in `profanityActions.js` to avoid error after checking its precondition
- fix `!afk` command and related actions

### Tests
- update outdated tests. This is a bigger project than I anticipated, but I did get started on it here
- add new tests to cover more commands
- update scripts in `package.json`

### Other changes
- make logs more readable by stopping Sequelize from logging so much
- remove unused lines in `.gitignore`
- actually ignore `data/cotw.json`
- remove various comments throughout codebase
- move redundant `client.on('ready')` function into `ready.js` (closes #247)
- move all database `sync` functions into one call in `ready.js`
- enforce code style (linter coming soon)
- change line endings in `firstweek.js` and `firstweek.test.js` from CRLF to LF
- add missing await statements for clarity when calling async functions
- move tests to a `test` folder

### CODE REVIEW

File: commands/firstweek.js [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
-
-module.exports.config = {
-	name: 'firstweek',
-	aliases: ['fw'],
-	description: 'I will fetch you the link to the First Week at KOA post.',
-	usage: ['firstweek']
-};
+module.exports.execute = async (client, message) => {
+	// Link enclosed in angle brackets to stop Discord from automatically embedding the link
+	return await message.channel.send('Your first week at KOA: <https://knightsofacademia.org/your-first-week-at-koa>');
```
Comment (ThatNerdSquared) [inline commands/firstweek.js:3]: Not exactly related to the code but the article at this "first week" link is pretty outdated, not sure if you wanted to remove this command for now or alert somebody on the website team about it.

File: commands/tests/article.test.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+			};
+		}
+	});
+});
+
+afterAll(() => {
+	jest.restoreAllMocks();
+});
+
+test('Sends correct response with no arguments', async() => {
```
Comment (ThatNerdSquared) [inline commands/tests/article.test.js:39]: Would it be a good idea to have a test to make sure misspelled arguments return some sort of error message to the user?

File: commands/tests/help.test.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+afterAll(() => {
+	client.destroy();
+});
+
+test('Sends default help message when no argument is provided', async () => {	
+	await help.execute(client, message, []);
+
+	const expectedResponse = 'I have sent you a private message with the command list.';
+	expect(message.author.send).toHaveBeenCalledTimes(1);
```
Comment (ThatNerdSquared) [inline commands/tests/help.test.js:36]: Do we need to test if the embed DMed to the user is correct?

File: databaseFiles/tosReminderTable.js [hunk 4]
```diff
@@ -4,18 +4,18 @@ const connect = require('./connect.js');
...
 
 module.exports = sequelize.define('TosReminder', {
-	user_id:{
+	user_id: {
         type: DataTypes.STRING,
         allowNull: false
     },
     joinTime: {
         type: DataTypes.DATE,
-        allowNull : false
```
Comment (ThatNerdSquared) [inline databaseFiles/tosReminderTable.js:13]: Could you fix the different indentations (tabs/spaces)?

File: eventActions/sleepClubActions.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 const config = require('../config.json');
 
-class sleepclubActions {
+class sleepClubActions {
 	static async reactToSleepLog(client, message) {
 		const emote = config.emotes.sleeplogReaction;
```
Comment (ThatNerdSquared) [inline eventActions/sleepClubActions.js:5]: I think this emote was removed, not sure what you want to do about that.

File: events/ready.js [hunk 6]
```diff
@@ -1,7 +1,15 @@
 const chainMessageAction = require('../eventActions/checkChainMessage');
+const config = require('../config.json');
+const { sequelize } = require('../databaseFiles/connect');
+const remind = require('../commands/remind');
+const tosRemind = require('../eventActions/tosReminderAction.js');
```
Comment (ThatNerdSquared) [inline events/ready.js:5]: A bit of a nitpick but most of the `require` statements here don't have the `.js` file extension whereas the `tosRemind` does, maybe remove the extension from all of them or add it for all of them.

**Review by ThatNerdSquared — CHANGES_REQUESTED** [review]

File: commands/firstweek.js (hunk 1 above)
Comment (debashisbiswas) [reply commands/firstweek.js:3]: Yep, that's on the website team. I'll bring it up, but that's not related to this PR specifically

File: commands/tests/article.test.js (hunk 2 above)
Comment (debashisbiswas) [reply commands/tests/article.test.js:39]: Added new functionality + test

File: commands/tests/help.test.js (hunk 3 above)
Comment (debashisbiswas) [reply commands/tests/help.test.js:36]: Added this to the test

File: databaseFiles/tosReminderTable.js (hunk 4 above)
Comment (debashisbiswas) [reply databaseFiles/tosReminderTable.js:13]: Fixed these with a bunch of other eslint fixes

File: eventActions/sleepClubActions.js (hunk 5 above)
Comment (debashisbiswas) [reply eventActions/sleepClubActions.js:5]: Removed the unused emote

File: events/ready.js (hunk 6 above)
Comment (debashisbiswas) [reply events/ready.js:5]: I standardized them here, but I'm hoping to add this to ESLint at some point.

**Review by ThatNerdSquared — APPROVED** [review]
Great job with all the fixes! 🎉

**Pull request merged**